bpf/gotracer/go_sql.c

```diff
@@ -26,10 +26,200 @@
 
 #include <gotracer/go_common.h>
 
-static __always_inline void set_sql_info(void *goroutine_addr, void *sql_param, void *query_len) {
+// Extracts MySQL server hostname by following the pointer chain:
+// driverConn -> ci (interface) -> mysqlConn -> cfg (*Config) -> Addr (string)
+// Returns the Addr string which has format "host:port"
+static __always_inline int
+read_mysql_hostname_from_driverconn(u64 driver_conn_ptr, char *hostname, int max_len) {
+    if (driver_conn_ptr == 0) {
+        return 0;
+    }
+
+    off_table_t *ot = get_offsets_table();
+    if (!ot) {
+        return 0;
+    }
+
+    // Get offset of driverConn.ci field
+    u64 ci_offset = go_offset_of(ot, (go_offset){.v = _driverconn_ci_pos});
+    if (ci_offset == (u64)-1) {
+        return 0;
+    }
+
+    // driverConn.ci is a Go interface [type_ptr (8 bytes), data_ptr (8 bytes)]
+    // Read the data pointer (at ci_offset + 8) to get the actual mysqlConn instance
+    void *mysql_conn_ptr = NULL;
+    bpf_probe_read(
+        &mysql_conn_ptr, sizeof(mysql_conn_ptr), (void *)(driver_conn_ptr + ci_offset + 8));
+
+    if (!mysql_conn_ptr) {
+        return 0;
+    }
+
+    // Read mysqlConn.cfg field offset
+    u64 cfg_offset = go_offset_of(ot, (go_offset){.v = _mysql_conn_cfg_pos});
+    if (cfg_offset == (u64)-1) {
+        return 0;
+    }
+
+    // Dereference mysqlConn.cfg to get pointer to Config struct
+    void *cfg_ptr = NULL;
+    bpf_probe_read(&cfg_ptr, sizeof(cfg_ptr), (void *)((u64)mysql_conn_ptr + cfg_offset));
+
+    if (!cfg_ptr) {
+        return 0;
+    }
+
+    // Read Config.Addr field offset
+    u64 addr_offset = go_offset_of(ot, (go_offset){.v = _mysql_config_addr_pos});
+    if (addr_offset == (u64)-1) {
+        return 0;
+    }
+
+    // Config.Addr is a Go string represented as [data_ptr, length]
+    u64 addr_ptr = 0, addr_len = 0;
+    bpf_probe_read(&addr_ptr, sizeof(addr_ptr), (void *)((u64)cfg_ptr + addr_offset));
+    bpf_probe_read(&addr_len, sizeof(addr_len), (void *)((u64)cfg_ptr + addr_offset + 8));
+
+    if (addr_ptr == 0 || addr_len == 0) {
+        return 0;
+    }
+
+    if (addr_len > max_len - 1) {
+        addr_len = max_len - 1;
+    }
+
+    bpf_probe_read(hostname, addr_len, (void *)addr_ptr);
+    if (addr_len < max_len) {
+        hostname[addr_len] = '\0';
+    }
+
+    // Basic validation: check first few chars are printable
+#pragma unroll
```

Comment:
Why do we want to do this validation here? Do we expect the value to be corrupt? If you have to do validation, it's better done in userspace when we create the event.